### PR TITLE
Migrate to unified build script

### PR DESCRIPTION
**Pull request by SinTh0r4s**
@Glease Do you want this upstream?

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      run: ./gradlew setupCIWorkspace
+    
+    - name: Build the mod
+      run: ./gradlew build
+
+    - name: Run server for 1 minute
+      run: |
+        mkdir run
+        echo "eula=true" > run/eula.txt  
+        timeout 10 ./gradlew runServer || true
```
Comment (Glease) [inline .github/workflows/build-and-test.yml]: since when is 1 minute 10 seconds?

**Review by Glease — CHANGES_REQUESTED** [review]

**Comment by Glease** [comment]
No, I don't need this in upstream.

**Comment by Glease** [comment]
I have jar signing, publish to cursforge in upstream and changelog generation, which is not in this unified buildscript. You can do whatever in GTNH fork but let's don't break my upstream publish workflow.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (SinTh0r4s) [reply .github/workflows/build-and-test.yml]: Typo in the master CI used for copying. Will be fixed with the planned mass update once all mods are migrated

**Comment by SinTh0r4s** [comment]
> I have jar signing, publish to cursforge in upstream and changelog generation, which is not in this unified buildscript. You can do whatever in GTNH fork but let's don't break my upstream publish workflow.

Was just an offer! ;-)
Also: We have the customization option to add a `addon.gradle` file to the repository. It will be integrated into the buildscript if present and can contain customization like signing and so on. This file will not be touched by updates. Just FYI

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**